website/source/docs/providers/vsphere/r/virtual_machine.html.markdown

```diff
@@ -113,6 +113,7 @@ The `disk` block supports:
 * `template` - (Required if size and bootable_vmdk_path not provided) Template for this disk.
 * `datastore` - (Optional) Datastore for this disk
 * `size` - (Required if template and bootable_vmdks_path not provided) Size of this disk (in GB).
+* `name` - (Required if size is provided) Name of the new disk file to create.
```

Comment:
Can we just generate a name if they don't give us one?  Or document what the name means :)